app/src/main/java/com/example/plantsapp/presentation/ui/notification/NotificationWorker.kt

```diff
@@ -0,0 +1,119 @@
+package com.example.plantsapp.presentation.ui.notification
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import androidx.hilt.work.HiltWorker
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.bumptech.glide.Glide
+import com.example.plantsapp.R
+import com.example.plantsapp.domain.model.Plant
+import com.example.plantsapp.domain.model.Task
+import com.example.plantsapp.domain.repository.TasksRepository
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
+import java.util.Date
+
+@HiltWorker
+class NotificationWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted workerParams: WorkerParameters,
+    private val repository: TasksRepository
+) : Worker(context, workerParams) {
+
+    private var notificationId = 0
+
+    override fun doWork(): Result {
+        runBlocking {
+            val plantsWithTasks =
+                repository.getPlantsWithTasksForDate(Date()).first()
+
+            plantsWithTasks.forEach { (plant, tasks) ->
+                sendTaskNotifications(plant, tasks)
+            }
+        }
+
+        return Result.success()
+    }
+
+    private fun sendTaskNotifications(
+        plant: Plant,
+        tasks: List<Task>
+    ) {
+        createChannel(context)
+
+        val notificationPicture = Glide.with(context)
+            .asBitmap()
+            .load(plant.plantPicture ?: R.drawable.ic_baseline_image_24)
```

Comment:
![image](https://user-images.githubusercontent.com/23121846/142023871-4ebe25ce-b1c8-4414-b21a-382537571659.png)
Also if you select image from camera - this falls